source/concurrent/PartiallyGuardedList.h

```diff
@@ -0,0 +1,59 @@
+/* PartiallyGuardedList.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef PARTIALLY_GUARDED_LIST_H_
+#define PARTIALLY_GUARDED_LIST_H_
+
+#include <list>
+#include <mutex>
+
+
+
+// Vectors that guard against most concurrent modifications. Use with caution.
+// For most purposes, this can be treated and passed around like any stl container,
+// but any function modifying it inside a concurrent context MUST receive it as a PartiallyGuarded object
+// to ensure proper handling.
+template<class T, class Allocator = std::allocator<T>>
+class PartiallyGuardedList : public std::list<T, Allocator> {
```

Comment:
would it maybe be easier to have PartiallyGuardedList  not inherit from std::list, but store an instance of it?
that way we only have to provide the functions we need and do not have to delete everything else

Reply:
No, because there are functions that I want to make available unguarded, and I also didn't want to touch any of the code where I pass these objects around.

Reply:
> because there are functions that I want to make available unguarded,

would be possible with that approach

> and I also didn't want to touch any of the code where I pass these objects around.

ok